app/src/main/java/com/nmakademija/nmaakademija/fragment/ScheduleFragment.java

```diff
@@ -84,9 +84,25 @@ private void setScheduleItems(List<ScheduleEvent> scheduleEvents) {
         }
 
         Collections.sort(scheduleItems, new ScheduleItemComparator());
+        int i = 0;
+        Date now = new Date();
+        while (i < scheduleItems.size()) {
+            if (scheduleItems.get(i).getDate().before(now))
+                i = i + 1;
+            else
+                break;
+        }
 
-        ScheduleAdapter adapter = new ScheduleAdapter(getContext(), scheduleItems);
+        ScheduleAdapter adapter = new ScheduleAdapter(scheduleItems);
         scheduleListView.setAdapter(adapter);
+        scheduleListView.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false));
+        scheduleListView.setHasFixedSize(false);
+        if (i < scheduleItems.size()) {
+            scheduleListView.scrollToPosition(i);
+            //TODO if(i+1==scheduleItems.size())    bottomnavigation.dontshow
+        } else {
+            Error.oldData(getView());
```

Comment:
WTF? If there is no upcoming events, then you shouldn't show error or do anything like this. Why did you add this function?